bundles/EcommerceFrameworkBundle/Resources/config/index_service_configs_workers.yml

```diff
@@ -67,15 +67,25 @@ services:
 
     Pimcore\Bundle\EcommerceFrameworkBundle\IndexService\Worker\OptimizedMysql:
         parent: Pimcore\Bundle\EcommerceFrameworkBundle\IndexService\Worker\AbstractWorker
+        arguments:
+            $workerMode: '%pimcore_ecommerce.index_service.worker_mode%'
 
     Pimcore\Bundle\EcommerceFrameworkBundle\IndexService\Worker\ElasticSearch\DefaultElasticSearch5:
         parent: Pimcore\Bundle\EcommerceFrameworkBundle\IndexService\Worker\AbstractWorker
+        arguments:
+            $workerMode: '%pimcore_ecommerce.index_service.worker_mode%'
```

Comment:
I'm not sure if this is relevant for BC. However, if you've got a service extending one of the existing workers (e.g. DefaultElasticSearch5) you'll now have to provide the argument $workerMode in the service definition, as `ProductCentricBatchProcessingWorker`'s constructor has new non-nullable `string $workerMode` as 4th argument and auto-wiring doesn't work anymore:

```
Fatal error: Uncaught Symfony\Component\DependencyInjection\Exception\RuntimeException: Cannot autowire service "pimcore_ecommerce.index_service.products.worker": argument "$workerMode" of method "Pimcore\Bundle\EcommerceFrameworkBundle\IndexService\Worker\ElasticSearch\AbstractElasticSearch::__construct()" is type-hinted "string", you should configure its value explicitly. in /var/www/pimcore6/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/Compiler/DefinitionErrorExceptionPass.php on line 54
```

Example definition to get it working again:

```yaml
services:
    # Index service worker
    AppBundle\EcommerceFramework\IndexService\Worker\Products:
        parent: Pimcore\Bundle\EcommerceFrameworkBundle\IndexService\Worker\AbstractWorker
        arguments:
            $workerMode: '%pimcore_ecommerce.index_service.worker_mode%'
```

Reply:
should be fixed, workerMode is optional now